packages/keyring-controller/src/KeyringController.ts

```diff
@@ -926,76 +929,74 @@ export class KeyringController extends BaseController<
    * operation completes.
    */
   async persistAllKeyrings(): Promise<boolean> {
-    const { encryptionKey, encryptionSalt } = this.state;
+    return this.#withVaultLock(async () => {
```

Comment:
Perhaps we could move the `persistAllKeyring` logic along with this vault lock to an internal private function, so that we can put a controller lock on the public `persistAllKeyrings` without having a deadlock, since the public function is currently called internally by many other methods